### ISSUE TITLE
gassertFailure not expanded by `htfpp`

### ISSUE THREAD

**Issue by league** [skogsbaer/HTF#63]
When I try to use `gassertFailure` in a source file processed by `htfpp`, it doesn't expand and thus generates a compiler error:

```
Not in scope: ‘gassertFailure’
Perhaps you meant one of these:
  ‘gassertFailure_’ (imported from Test.Framework),
  ‘assertFailure_’ (imported from Test.Framework),
  data constructor ‘WaiTestFailure’ (imported from Network.Wai.Test)
```

In `Test.Framework.Preprocessor`, I see that `allAsserts` is missing `assertFailure` but it's patched in later, after adding the `g` variants.

_As an aside,_ the reason I'm using the generic assertions is not for a pure result, but rather I want to use them within a monad stack which instantiates `MonadIO`. So I defined this orphan instance:

``` haskell
instance AssertM (ReaderT r IO) where
  genericAssertFailure__ loc s = liftIO $ genericAssertFailure__ loc s
  genericSubAssert loc ms act = do
    r <- ask
    liftIO $ genericSubAssert loc ms $ runReaderT act r
```

Would there be any interest in supporting `MonadIO` explicitly, either by defining similar `AssertM` instances, or by generalizing the regular `assert*` functions using `MonadIO`?

**Comment by skogsbaer** [comment]
I don't like the idea of generalizing the `assert*` functions to `MonadIO`. Error messages tend to get confusing this way. Which instances for `AssertM` do you need (apart from the `ReaderT` example?)

**Comment by skogsbaer** [comment]
The problem with gassertFailure not being expanded should be addressed with #72

**Comment by skogsbaer** [comment]
#72 is now implemented in #115 (available with version 0.15.0.0)
